### DISCUSSION TITLE
update canvas with changeprop in input

### DISCUSSION THREAD

**Discussion by tatan-pelaez** [GrapesJS/grapesjs#4585] [General]
hello, good afternoon, what happens is that grapes is incorporated into my appweb to generate forms.

The thing is that I already created the form, I have a changeprop to an input to make automatic formulas between inputs. everything stores great, works fine

The problem occurs when I want to update that form, I already make the form appear on the canvas, but the changeprop does not take me to the input.

I have tried any number of things that occur to me, but I do not achieve anything, I appreciate a help

this is my DomComponents input
```
editor.DomComponents.addType('input', {
    isComponent: el => el.tagName == 'INPUT',
    model: {
      defaults: {
        script,
        stitle: "",
        'script-props': ['stitle'],
        traits: [
          'name',
          'placeholder',
          {
            type: 'text',
            name: "stitle",
            changeProp: true,
            label: 'Formula',
          },
          {
            type: 'select',
            label: 'Type',
            name: 'type',
            options: [
              { id: 'text', name: 'Text'},
              { id: 'email', name: 'Email'},
              { id: 'password', name: 'Password'},
              { id: 'number', name: 'Number'},
            ]
          },
          {
            type: 'checkbox',
            name: 'required',
          }
        ],
        attributes: { id: 'campos', type: 'text'},
      },
    },
});
```

I store the form in mysql and in this way I incorporate it

`editor.setComponents(htmlcss);`

for the changeprop I only store this portion in mysql, which is the prop

`var props = {"campos":{"stitle":""},"campos-2":{"stitle":""},"campos-3":{"stitle":"a+b"},"campos-4":{"stitle":""}};`

thanks

**Comment by artf** [comment]
Hi @tatan-pelaez I'm not sure if I understand properly your issue but if you encounter issues on the load of your custom component, be sure to load the JSON of your project, not the html/css https://grapesjs.com/docs/modules/Storage.html#project-data
